Property Manager - Review and Confirm, App & Reg Details pages

dimak1 opened the pull request:
*Issue:*

- bcgov/entity/issues/23745
- bcgov/entity/issues/23746

*Description of changes:*
- Add Property Manager section to Review page
- Add Property manager section to Application Details and Registration Details pages
- Add unit tests
- Small fixes and minor tech debt cleanup

![Screenshot 2024-10-25 at 09 57 11](https://github.com/user-attachments/assets/9a59c782-096c-4689-b9c7-b503cbbfccf9)


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namex license (Apache 2.0).

dimak1 commented:
Pending due to PR #209

dimak1 commented:
/gcbrun

bcregistry-sre commented:
Temporary Url for review: https://strr-ui-dev--pr-210-f3c37s3c.web.app

shaangill025 commented:
> The fields in the detail pages are empty

@dimak1 , I believe adding `propertyManager` to `strr-web/interfaces/application-i.ts` will fix it [below].
```
export interface ApplicationDetailsI {
  registrationType: RegistrationTypeE,
  listingDetails: ListingDetailsI[]
  primaryContact: ContactI
  secondaryContact?: ContactI
  principalResidence: PrincipalResidenceI
  unitAddress: RegistrationAddressI
  propertyManager?: PropertyManagerI
  unitDetails: {
    parcelIdentifier?: string
    businessLicense?: string
    businessLicenseExpiryDate?: string
    propertyType: string
    ownershipType: string
  }
  documents?: DocumentUploadI[]
}
```

dimak1 commented:
Merged related PR. Rebased with main. Fixed issues and tests. Submission works, as well as Review, App Details and Reg Details pages. 
Pretty much got the Property Manager working. Please re-review. Thanks!
@shaangill025 @kris-daxiom @rstens

dimak1 commented:
/gcbrun

bcregistry-sre commented:
Temporary Url for review: https://strr-ui-dev--pr-210-f3c37s3c.web.app

shaangill025 commented on strr-web/interfaces/application-i.ts:
```diff
@@ -44,6 +44,7 @@ export interface ApplicationDetailsI {
   primaryContact: ContactI
   secondaryContact?: ContactI
   principalResidence: PrincipalResidenceI
+  propertyManager: PropertyManagerI
```
I think propertyManager should be optional

dimak1 replied:
Good catch, yea it should be optional.

dimak1 replied:
Updated

shaangill025 commented on strr-web/interfaces/registration-i.ts:
```diff
@@ -9,31 +9,32 @@ export interface RegistrationAddressI {
+  primaryContact: ContactI
+  secondaryContact: ContactI | null
+  principalResidence: PrincipalResidenceI
+  propertyManager: PropertyManagerI
```
Same as above, propertyManager should be optional?

dimak1 replied:
Updated

dimak1 commented:
/gcbrun

bcregistry-sre commented:
Temporary Url for review: https://strr-ui-dev--pr-210-f3c37s3c.web.app

shaangill025 reviewed (approved):
LGTM

dimak1 commented on strr-web/components/bcros/form-section/contact-information/Form.vue:
```diff
@@ -203,13 +203,13 @@ onMounted(() => {
 
 watch(form, () => {
-  if (form.value && isComplete) { form.value.validate() }
+  if (form.value && isComplete) { form.value.validate({ silent: true }) }
```
@shaangill025 this removed the validation error in console and fixed the issue with tests failures.
